app/Http/Middleware/LimitWikiAccess.php

```diff
@@ -10,21 +10,32 @@
 class LimitWikiAccess
 {
     /**
-     * Handle an incoming request.
-     *
-     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     * Reject any incoming request unless the user is a manager of the
+     * requested wiki. If the user is authorized, inject the wiki
+     * object into the request context.
      */
     public function handle(Request $request, Closure $next): Response
     {
-        $userHasAccess = WikiManager::where([
+        $validatedInput = $request->validate([
+            'wiki' => ['required', 'integer']
+        ]);
+
+        $wikiManager = WikiManager::where([
             'user_id' => $request->user()?->id,
-            'wiki_id' => $request->input('wiki'),
-        ])->exists();
+            'wiki_id' => $validatedInput['wiki'],
+        ])
+        ->with('wiki')
+        ->first();
 
-        if (!$userHasAccess) {
+        if (!$wikiManager) {
             abort(403);
         }
 
+        if (!$wikiManager->wiki) {
```

Comment:
I suspect this block should be higher in the middleware; for any random non-existing wiki I think this middleware currently returns a 403 not 404